controllers/machinehealthcheck_controller.go

```diff
@@ -274,7 +327,32 @@ func (r *MachineHealthCheckReconciler) reconcile(ctx context.Context, logger log
 			t.string(),
 		)
 	}
+
 	for _, t := range healthy {
+		if m.Spec.RemediationTemplate != nil {
+			remediationObj := &corev1.ObjectReference{
+				APIVersion: m.Spec.RemediationTemplate.APIVersion,
+				Kind:       strings.TrimSuffix(m.Spec.RemediationTemplate.Kind, external.TemplateSuffix),
+				Name:       t.Machine.Name,
+			}
+
+			// Get remediation request object
+			obj, err := external.Get(ctx, r.Client, remediationObj, t.Machine.Namespace)
+			if err != nil {
+				if apierrors.IsNotFound(errors.Cause(err)) {
+					continue
+				}
+				return ctrl.Result{}, errors.Wrapf(err, "failed to fetch remediation request for machine %q in namespace %q within cluster %q", t.Machine.Name, t.Machine.Namespace, t.Machine.ClusterName)
```

Comment:
This could be a potential early return from the loop without processing every element in `healthy`. I would suggest accumulating errors but not returning early.

Reply:
@ncdc is this ok now?